greeter_plugin/greeter_summary.py

```diff
@@ -0,0 +1,94 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Simple demo which greets several people."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+"""This module provides summaries for the Greeter plugin."""
+
+import tensorflow as tf
+
+
+PLUGIN_NAME = 'greeter'
+
+
+def op(name,
+       guest,
+       display_name=None,
+       description=None,
+       collections=None):
+  """Create a TensorFlow summary op to greet the given guest.
+
+  Arguments:
+    name: A name for this summary operation.
+    guest: A rank-0 string `Tensor`.
+    display_name: If set, will be used as the display name
+      in TensorBoard. Defaults to `name`.
+    description: A longform readable description of the summary data.
+      Markdown is supported.
+    collections: Which TensorFlow graph collections to add the summary
+      op to. Defaults to `['summaries']`. Can usually be ignored.
+  """
+
+  # The `name` argument is used to generate the summary op node name.
+  # That node name will also involve the TensorFlow name scope.
+  # By having the display_name default to the name argument, we make
+  # the TensorBoard display clearer.
+  if display_name is None:
+    display_name = name
+
+
+  summary_metadata = tf.SummaryMetadata()
+  # We could put additional metadata other than the PLUGIN_NAME,
+  # but we don't need any metadata for this simple example.
+  summary_metadata.plugin_data.add(plugin_name=PLUGIN_NAME, content="")
+  message = tf.string_join(['Hello, ', guest, '!'])
+  # Return a summary op that is properly configured.
+  return tf.summary.tensor_summary(
+    name,
+    message,
+    display_name=display_name,
+    summary_metadata=summary_metadata,
+    summary_description=description,
+    collections=collections)
+
+
+def pb(tag, guest, display_name=None, description=None):
+  """Create a greeting summary for the given guest.
+
+  Arguments:
+    tag: The string tag associated with the summary.
+    guest: The string name of the guest to greet.
+    display_name: If set, will be used as the display name in
+      TensorBoard. Defaults to `tag`.
+    description: A longform readable description of the summary data.
+      Markdown is supported.
+    """
+  message = 'Hello, %s!' % guest
+  tensor = tf.make_tensor_proto(message, dtype=tf.string)
+
+  summary_metadata = tf.SummaryMetadata(display_name=display_name,
+                                        summary_description=description)
+  metadata_content = '{}'  # We have no metadata to store.
```

Comment:
Maybe this comment could further clarify:

We have no metadata to store, but we add a plugin data proto anyway in order to associate this summary with the greeter plugin.